test/spec/Feature/Query/PgSafeUpdateSpec.hs

```diff
@@ -0,0 +1,49 @@
+module Feature.Query.PgSafeUpdateSpec where
+
+import Network.Wai (Application)
+
+import Network.HTTP.Types
+import Test.Hspec          hiding (pendingWith)
+import Test.Hspec.Wai
+import Test.Hspec.Wai.JSON
+
+import Protolude  hiding (get, put)
+import SpecHelper
+
+spec :: SpecWith ((), Application)
+spec =
+  describe "Enabling pg-safeupdate" $ do
+    context "Full table update" $ do
+      it "does not update and throws no error if no condition is present" $
+        request methodPatch "/safe_update"
+            [("Prefer", "tx=commit"), ("Prefer", "count=exact")]
+            [json| {"name": "New name"} |]
+          `shouldRespondWith`
+            ""
+            { matchStatus  = 404
+            , matchHeaders = [ matchHeaderAbsent hContentType
+                             , "Content-Range" <:> "*/0" ]
+            }
```

Comment:
So this proves that `pgsafeupdate` is broken for full table updates. It should throw a `UPDATE requires a WHERE clause`.

Related https://github.com/PostgREST/postgrest/pull/2195#discussion_r932946412